src/projekt_zaliczeniowy.Infrastructure/Data/Repositories/RestaurantsRepository.cs

```diff
@@ -17,9 +17,17 @@ public RestaurantsRepository(ApplicationDbContext context)
     this._context = context;
   }
 
-  public IEnumerable<Restaurant> getAllRestaurants()
+  public IEnumerable<Restaurant> getAllRestaurants(string? City, string? Region, string? Type)
   {
-    return _context.Restaurants.Include(a => a.CityNavigation).Include(a => a.TypeNavigation).ToList();
+
+    return _context.Restaurants
+     .Include(a => a.CityNavigation)
+     .Include(a => a.TypeNavigation)
+     .ThenInclude(t => t != null ? t.RegionNavigation : null)
+     .Where(a => City == null || (a.CityNavigation != null && a.CityNavigation.Name == City))
```

Comment:
ja bym zrobił `if(city is not null) query = query.Where(...)` - bardziej czytelne, ale dopóki to działa to też może zostać

Reply:
ok, bede pamietal